src/errorpage.cc

```diff
@@ -1248,25 +1328,127 @@ ErrorState::BuildContent()
+            const char *t = convert(++p, building_deny_info_url, allowRecursion);
+            result.appendf("%s", t);
+            m = p + 1;
+        } else {
+            assert(*p == '@');
+            handleLogFormat(p, result);
+            m = p;
+        }
     }
 
     if (*m)
-        content->appendf("%s", m);   /* copy tail */
+        result.appendf("%s", m);   /* copy tail */
+}
 
-    content->terminate();
+void
+ErrorState::handleLogFormat(const char *&start, MemBuf &result)
+{
+    assert(LogFormatStart.cmp(start, LogFormatStart.length()) == 0);
+
+    const char *p = start + LogFormatStart.length();
+    if (int tokenBytes = Format::Format::AssembleToken(p, result, al)) {
+        p += tokenBytes;
+        if (*p != '}') {
+            if (errorHandler_)
+                errorHandler_->handleError(SBuf("unterminated @Squid{"));
+            start = p;
+        } else
+            start = p + 1;
+        return;
+    }
 
-    assert((size_t)content->contentSize() == strlen(content->content()));
+    if (errorHandler_)
+        errorHandler_->handleError(ToSBuf("Cannot parse @Squid{logformat} in error page template near '", SBuf(start).substr(0, 100), "'"));
+
+    // Try recover to continue parsing
+    // The p points after '@Squid{'. Try to find the '}'.
+    if (const char *endOfComment = strchr(p, '}'))
+        p = endOfComment + 1;
+    else
+        while (*p == '%') ++p; // just skip the first '%' after '@Squid{'
+    result.append(start, p - start);
+    start = p;
+}
 
-    return content;
+ErrTextValidator &
+ErrTextValidator::useCfgContext(const char *filename, int lineNo, const char *line)
+{
+    ctx = CtxConfig;
+    ctxFilename = filename;
+    ctxLineNo_ = lineNo;
+    ctxLine_ = line;
+    return *this;
+}
+
+ErrTextValidator &
+ErrTextValidator::useFileContext(const char *templateFilename)
+{
+    ctx = CtxFile;
+    ctxFilename = templateFilename;
+    return *this;
+}
+
+bool
+ErrTextValidator::validate(const char *text)
+{
+    Must (initialised());
+
+    MemBuf content;
+    content.init();
+    AccessLogEntry::Pointer alp= new AccessLogEntry;
+    ErrorState anErr(ERR_NONE, Http::scNone, NULL, alp);
+    SBuf prefix;
+    if (ctx == CtxConfig)
+        prefix = ToSBuf(name_ , ": " ,  ctxFilename , " line " , ctxLineNo_ , ": " , ctxLine_);
+    else if (ctx == CtxFile)
+        prefix = ToSBuf(name_, ": Error while parsing file ",  ctxFilename);
+
+    ErrorState::ErrorHandler *handler;
+    if (onError_ == doReport)
+        handler = new ReportErrorHandler(warn_, name_, prefix);
+    else if (onError_ == doQuit)
+        handler = new FatalErrorHandler(warn_, name_, prefix);
+    else
+        handler = new ThrownErrorHandler(warn_, name_, prefix);
+
+    anErr.errorHandler_.reset(handler);
+
+    // Any possible thrown exceptions should handled from the caller.
+    anErr.convertAndWriteTo(text, content, (ctx == CtxConfig), true);
+
+    return onError_ == doReport ? true : (handler->errors() != 0);
```

Comment:
The validate() return true or false if the template can be used or not. It may print warnings but finally may allow or not  the use of the template. 
It is like the compilers, in some cases the warnings are errors or just warnings.